version: "3.8"

services:
  db:
    image: postgres:15
    environment:
      POSTGRES_USER: db
      POSTGRES_PASSWORD: hehe123
      POSTGRES_DB: monitoring
    volumes:
      - pgdata:/var/lib/postgresql/data
    ports:
      - "5434:5432"   
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5 
    networks:
      monitoring:    

  pgadmin:
    image: dpage/pgadmin4
    container_name: pgadmin4_container
    restart: always
    ports:
      - "8888:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: userhehe@hehe.com
      PGADMIN_DEFAULT_PASSWORD: hehe123
    volumes:
      - pgadmin:/var/lib/pgadmin
    networks:
      - monitoring
    

  backend:
    build: ./agent
    ports:
      - "8010:8000"  
    environment:
      DB_HOST: db
      DB_PORT: 5432
      DB_USER: db
      DB_PASSWORD: hehe123
      DB_NAME: monitoring
    depends_on:
      db:
        condition: service_healthy
    restart: always
    networks:
      monitoring:

  grafana:
    image: grafana/grafana:7.5.7
    ports:
      - 3010:3000
    environment:
      - GF_PATHS_CONFIG=/etc/grafana/grafana.ini
      - GF_PATHS_PROVISIONING=/etc/grafana/provisioning
    volumes:
      - grafana-data:/var/lib/grafana
      - ./grafana:/etc/grafana
    dns:
      - 8.8.8.8
      - 8.8.4.4
    networks:
      monitoring:
    extra_hosts:
      - "host.docker.internal:host-gateway"



volumes:
  pgadmin:
  pgdata:
  prometheus_data:
  grafana-data:

networks:
  monitoring:
    external: true
